### PR TITLE
WebAPI for TLD users

### PR DESCRIPTION
**Pull request by Sandmann79**
Hi @Varstahl you may noticed that i'm working on joining TLD & Primevideo together, so it's available for all users.
The approach was to use it on a single module, to later easily add new features or fix bugs for both ends.

There some added/changed features only available for TLDs (_cursive in the list below_), to avoid breaking too much things for PV :wink:

Here're the changes done:
* _added the _mmlinks appendix to split the Categories section into submenus, linke on website. Without that on TLD there 37 entries at the main menu._
* _added basic Watchlist editing. Basic means, if you're inside the Watchlist folder, you can remove items, everywhere else you can add items. There's no check if an item is in the Watchlist. 
To get an actual token the [/gp/video/api/enrichItemMetadata](www.primevideo.com/gp/video/api/enrichItemMetadata) url is requested, which is also available for PV.
You can enable it at [this line](https://github.com/Sandmann79/xbmc/blob/3457f2b7f42b1f0a024c3aec70d9e60ca8765f11/plugin.video.amazon-test/resources/lib/web_api.py#L638)_
* MyStuff section in the main menu, displays the same content as on the Webpage (Watchlist, Purchased...)
* changed sorting of items as on the Webpage, by adding a `pos` key to the metadata dict. This disables also the alphabetic sorting of movie/tvshows/one-season pages
* _always paginate collections
Preloads 2, if the setting is enabled else 10 pages. Due a wrong placed import statement at `GrabJSON `method, the parsing crashed at a different page, but after fixing this there are some sections on TLD, which contains 50+ pages. To avoid large loading times, this limitation has been introduced [here](https://github.com/Sandmann79/xbmc/blob/3457f2b7f42b1f0a024c3aec70d9e60ca8765f11/plugin.video.amazon-test/resources/lib/web_api.py#L1426)._
* use binary download, if `/api/` is in the url
This is 6 times faster as without, but only on these pages, which returns only json.
Proof:
```
https://www.amazon.de/gp/video/api/storefront/ref=atv_hm_nxt_2?startIndex=7&targetId=V2%3D4AEA6u69gYPeuYe-toZwYWdlSWSIcGFnZVR5cGWMY29sbGVjdGlvbklkiHdpZGdldElkjo5zd2lmdElkVmVyc2lvbt6UioRob21li4Rob21ljA-ND46CVjI%3D&pageSize=14&contentId=home&contentType=home&pageNumber=2&clientFilter=on&serviceToken=&episodeListSize=9999
2021-02-04 10:10:45.504 T:5628  NOTICE: [Amazon VOD] network.py:225 Download Time: 1.327123
2021-02-04 10:10:45.505 T:5628  NOTICE: [Amazon VOD] network.py:144 getURL: https://www.amazon.de/gp/video/api/storefront/ref=atv_hm_nxt_2?startIndex=7&targetId=V2%3D4AEA6u69gYPeuYe-toZwYWdlSWSIcGFnZVR5cGWMY29sbGVjdGlvbklkiHdpZGdldElkjo5zd2lmdElkVmVyc2lvbt6UioRob21li4Rob21ljA-ND46CVjI%3D&pageSize=14&contentId=home&contentType=home&pageNumber=2&clientFilter=on&serviceToken=&episodeListSize=9999
2021-02-04 10:10:55.587 T:5628  NOTICE: [Amazon VOD] network.py:225 Download Time: 10.0807493
2021-02-04 10:10:56.328 T:5628  NOTICE: [Amazon VOD] network.py:144 getURL: 
```
* added plot and actors to TVShow metadata
* renamed files and used the same name for `PrimeVideo` class
* added some translations
* fixed issues: #516, #531, #534
* some more i forget in the meanwhile…

If you have any questions, feel free to ask.

### CODE REVIEW

**Comment by Varstahl** [comment]
Looks great :)

I'll try to find the time to go through all of this in the weekend, there are definitely things I can introduce in PV to re-align the projects.

**Comment by Sandmann79** [comment]
To say it in your words - no need to rush :grin:

**Review by luizoti — APPROVED** [review]

**Comment by Sandmann79** [comment]
@luizoti i assume you're a primevideo user (italy?) and watchlist editing / category listing works for you with these patches?

**Comment by luizoti** [comment]
> @luizoti i assume you're a primevideo user and watchlist editing / category listing works for you with these patches?

Sorry, I accidentally approved in VSCode and didn't find option to remove approval.

Regarding this path, when opening a Watchlist, I don't remember getting errors, like other problems, I forgot to test.

The only problem I'm sure I'm trying to do is that the reproduction doesn't happen, but I think it's just me, because it happens in the beta version and in the version available in the kodi repository (that's why I didn't even open an issue about).

**Comment by luizoti** [comment]
> @luizoti i assume you're a primevideo user and watchlist editing / category listing works for you with these patches?

hi, i did new tests, activated the 'web api' option.

The refresh option did not return any error, but the information was not actually updated either, I say this because I still have problems with some series that X or Y season appear empty, without episodes and the refresh uction does not make these items appear, as mentioned [here](https://github.com/Sandmann79/xbmc/issues/534#issuecomment-767595126).

File: plugin.video.amazon-test/resources/settings.xml [hunk 1]
```diff
@@ -27,11 +27,12 @@
...
+        <setting id="enable_uhd" type="bool" label="UltraHD (Android + Widevine L1)" visible="eq(-20,3)" enable="System.Platform.Android" default="false"/>
+        <setting id="enable_hdr10" type="bool" label="HDR10+ (Android + Widevine L1)" visible="eq(-21,3)" enable="System.Platform.Android" default="false"/>
+        <setting id="enable_dovi" type="bool" label="Dolby Vision (Android + Widevine L1)" visible="eq(-22,3)" enable="System.Platform.Android" default="false"/>
+        <setting id="proxy_mpdalter" type="bool" label="30248" default="true" visible="eq(-23,3)" />
+        <setting id="skip_scene" type="enum" label="30081" lvalues="30082|30083|30084|30085|30086" default="1" visible="eq(-24,3)"/>
+        <setting id="send_vp" type="slider" label="30066" range="0,2,90" default="4" option="percent" visible="eq(-25,3)"/>
         <!-- Browser/Script -->
         <setting id="remotectrl" type="bool" label="30074" default="false" visible="lt(-25,2)"/>
         <setting id="remote_vol" type="bool" label="30075" default="false" subsetting="true" enable="eq(-1,true)" visible="lt(-26,2)"/>
```
Comment (Acidzero2020) [inline plugin.video.amazon-test/resources/settings.xml:38]: I get these error messages.

`cannot reference separator setting in old setting condition "eq(-25,2)" for "remotectrl"`
`failed to parse visible condition "eq(-25,2)" of old setting definition for "remotectrl"`
`cannot reference separator setting in old setting condition "eq(-26,2)" for "remote_vol"`
`failed to parse visible condition "eq(-26,2)" of old setting definition for "remote_vol"`

```suggestion
        <setting id="remotectrl" type="bool" label="30074" default="false" visible="lt(-26,2)"/>
        <setting id="remote_vol" type="bool" label="30075" default="false" subsetting="true" enable="eq(-1,true)" visible="lt(-27,2)"/>
```

Comment (Sandmann79) [reply plugin.video.amazon-test/resources/settings.xml]: thx for reporting - overread this

**Review by Varstahl — APPROVED** [review]
Tested it (not too thoroughly), and everything is working fine. There are some functionality regressions such as the "as in the website" results, which is the topmost reason why I started developing the addon as I absolutely *hated* how amazon is returning results, but it's in no way blocking. I'll add options to disable 1:1 mapping to the website later, if it's not already implemented.

Took me way too much time, but at least I made it in the end, that's what counts, right? :D

**Edit:** @Sandmann79 I'm not merging it right away because I haven't kept up with the discussions so I don't know if this is release ready. I did fix a few merge conflicts, though, so it should be good to go.

**Comment by Varstahl** [comment]
Pushed a couple fixes and a new release, tested up to Kodi 20, time to merge.

**Pull request merged**